docs/extension-system/viewer-editor-apps.md

```diff
@@ -0,0 +1,135 @@
+
+### App setup
+
+Inside the `src` folder you will need an `index.ts` file that sets up the app so it can be registered by the Web runtime. It follows the basic structure as described in [the apps section]({{< ref "_index.md#apps" >}}), so it may look like this:
+
+```typescript
+import { AppWrapperRoute, defineWebApplication } from '@ownclouders/web-pkg'
+import translations from '../l10n/translations.json'
+import { useGettext } from 'vue3-gettext'
+
+// This is the base component of your app.
+import App from './App.vue'
+
+export default defineWebApplication({
+  setup() {
+    // The ID of your app.
+    const appId = 'advanced-pdf-viewer'
+
+    const { $gettext } = useGettext()
+
+    // This creates a route under which your app can be opened.
+    // Later, this route will be bound to one or more file extensions.
+    const routes = [
+      {
+        name: 'advanced-pdf-viewer',
+        path: '/:driveAliasAndItem(.*)?',
+        component: AppWrapperRoute(App, {
+          applicationId: appId
+        }),
+        meta: {
+          authContext: 'hybrid',
+          title: $gettext('Advanced PDF Viewer'),
+          patchCleanPath: true
+        }
+      }
+    ]
+
+    return {
+      appInfo: {
+        name: 'Advanced PDF Viewer',
+        id: appId,
+        defaultExtension: 'pdf',
+        isFileEditor: true,
+        extensions: [
+          // This makes sure all files with the "pdf" extension will be routed to your app when being opened.
+          // See the `ApplicationFileExtension` interface down below for a list of all possible properties.
+          {
+            extension: 'pdf',
+            routeName: 'advanced-pdf-viewer',
+
+            // Add this if you want your app to be present in the "New" file menu.
+            newFileMenu: {
+              menuTitle() {
+                return $gettext('PDF document')
+              }
+            }
+          }
+        ],
+
+        // Add this if you want your app to be present in the app menu on the top left.
+        applicationMenu: {
+          enabled: () => true,
+          openAsEditor: true,
+          priority: 10
+        }
+      },
+      routes,
+      translations
+    }
+  }
+})
+```
+
+Here is the interface defining the `extensions` property of the `appInfo` object.
+
+```typescript
+interface ApplicationFileExtension {
+  app?: string
+  extension?: string
+  createFileHandler?: (arg: {
+    fileName: string
+    space: SpaceResource
+    currentFolder: Resource
+  }) => Promise<Resource>
+  hasPriority?: boolean
+  label?: string
+  name?: string
+  icon?: string
+  mimeType?: string
+  newFileMenu?: { menuTitle: () => string }
+  routeName?: string
+}
+```
+
+### Integration in Web
```

Comment:
For integration into web we now have what @fschade built. It's not available in oCIS 5.0.0, but will be in 5.1.0 / 6.0.0 (whatever comes next). A link to https://owncloud.dev/services/web/#web-apps should be sufficient instead of having this integration section.

The sentence at the end about developing from within the monorepo is a good addition though, at least until we have something better for dev setups of third party apps.

TL;DR: I'd remove this section again and place a link to the web service docs. But probably better in the `index.md` page since it applies for both third party extensions and apps. What do you think?